enderio-base/src/main/java/com/enderio/base/common/capability/darksteel/EnergyDelegator.java

```diff
@@ -0,0 +1,72 @@
+package com.enderio.base.common.capability.darksteel;
+
+import com.enderio.base.common.capability.EIOCapabilities;
+import com.enderio.base.common.item.darksteel.upgrades.EmpoweredUpgrade;
+import com.enderio.core.common.capability.MultiCapabilityProvider;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.Tag;
+import net.minecraftforge.common.util.INBTSerializable;
+import net.minecraftforge.energy.EnergyStorage;
+import net.minecraftforge.energy.IEnergyStorage;
+
+import java.util.Optional;
+
+public class EnergyDelegator implements IEnergyStorage , INBTSerializable<Tag>  {
+
+    private final MultiCapabilityProvider prov;
+
+    private static final EnergyStorage NULL_DELEGATE = new EnergyStorage(0);
+
+    public EnergyDelegator(MultiCapabilityProvider prov) {
+       this.prov = prov;
+    }
+
+    @Override
+    public int receiveEnergy(int maxReceive, boolean simulate) {
+        return getDelegate().receiveEnergy(maxReceive, simulate);
+    }
+
+    @Override
+    public int extractEnergy(int maxExtract, boolean simulate) {
+        return getDelegate().extractEnergy(maxExtract, simulate);
+    }
+
+    @Override
+    public int getEnergyStored() {
+        return getDelegate().getEnergyStored();
+    }
+
+    @Override
+    public int getMaxEnergyStored() {
+        return getDelegate().getMaxEnergyStored();
+    }
+
+    @Override
+    public boolean canExtract() {
+        return getDelegate().canExtract();
+    }
+
+    @Override
+    public boolean canReceive() {
+        return getDelegate().canReceive();
+    }
+
+    @Override
+    public Tag serializeNBT() {
+        return new CompoundTag();
+    }
+
+    @Override
+    public void deserializeNBT(Tag nbt) {}
```

Comment:
What is the purpose of having this implement INBTSerializable given it effectively is NO-OP'ing the methods?

Reply:
I thought capabilities had to be. I now know otherwise, thanks.